src/sentry/api/endpoints/team_all_unresolved_issues.py

```diff
@@ -80,19 +65,61 @@ def calculate_unresolved_counts(
         .annotate(open=Count("id"))
     )
 
+    # Pull extra data to do deduplication in Python. (Inefficient to do in SQL via subqueries
+    # (see ISWF-549); cannot do via DISTINCT ON state because of Django limitations.)
     bucketed_issues = (
         GroupHistory.objects.filter_to_team(team)
-        .filter(group_history_environment_filter, date_added__gte=start, date_added__lte=end)
-        .annotate(bucket=TruncDay("date_added"), prev_status=prev_status_sub_qs)
-        .filter(dedupe_status_filter)
-        .order_by("bucket")
-        .values("project", "bucket")
+        .filter(
+            group_history_environment_filter,
+            date_added__gte=start,
+            date_added__lte=end,
+            status__in=OPEN_STATUSES + CLOSED_STATUSES,
+        )
         .annotate(
-            open=Count("id", filter=Q(status__in=OPEN_STATUSES)),
-            closed=Count("id", filter=Q(status__in=CLOSED_STATUSES)),
+            bucket=TruncDay("date_added"),
+            state=Case(
+                When(status__in=OPEN_STATUSES, then=Value("open")),
+                When(status__in=CLOSED_STATUSES, then=Value("closed")),
+                default=Value("other"),
+            ),
+        )
+        .order_by(
+            "group_id",
+            "bucket",
         )
+        .values("project", "group_id", "bucket", "state")
     )
 
+    most_recent_group_state: defaultdict[str, str] = defaultdict(lambda: "other")
+    # Project => Bucket => State => Count
+    deduping_map: defaultdict[str, defaultdict[str, defaultdict[str, int]]] = defaultdict(
+        lambda: defaultdict(lambda: defaultdict(int))
+    )
+    for r in bucketed_issues:
+        # Don't process the row if it doesn't set the group to open or closed.
+        if r["state"] == "other":
+            continue
```

Comment:
Just related to the comment above, maybe not necessary since no rows can have this status?